### PR TITLE
Migration to spring3/jakarta/java17

### PR DESCRIPTION
**Pull request by alexgordtop**

### CODE REVIEW

**Comment by FrankSchnicke** [comment]
Closes https://github.com/eclipse-basyx/basyx-java-server-sdk/issues/113

**Review by mdanish98 — CHANGES_REQUESTED** [review]
Following are some of the review remarks:

**Comment by alexgordtop** [comment]
@mdanish98 I can't activate the kafka modules, since they aren't spring3 ready - in their current configuration. @FrankSchnicke and @geso02 agreed on this and @geso02 will contribute related changes as soon as this pull request is merged.

**Comment by mdanish98** [comment]
@alexgordtop yes sure I am removing those remarks.

File: basyx.common/basyx.http/pom.xml [hunk 1]
```diff
@@ -16,7 +16,18 @@
 			<groupId>org.eclipse.digitaltwin.basyx</groupId>
 			<artifactId>basyx.core</artifactId>
 		</dependency>
-
+		<dependency>
+			<groupId>jakarta.validation</groupId>
+			<artifactId>jakarta.validation-api</artifactId>
+		</dependency>
+		<dependency>
```
[the remark this answers is not in the document]
Comment (alexgordtop) [reply basyx.common/basyx.http/pom.xml:23]: Now it's required - for e.g. Base64UrlEncodedIdentifierSizeValidator

**Review by mdanish98 — APPROVED** [review]
Approved

**Comment by FrankSchnicke** [comment]
Thanks a lot!

**Pull request merged**